src/payment/strategies/masterpass/masterpass-payment-strategy.spec.ts

```diff
@@ -0,0 +1,257 @@
+    let masterpassScript: Masterpass;
+
+    beforeEach(() => {
+        initOptions = {
+            methodId: 'masterpass',
+        };
+
+        requestSender = createRequestSender();
+        orderRequestSender = new OrderRequestSender(createRequestSender());
+
+        store = createCheckoutStore({
+            checkout: getCheckoutState(),
+            config: getConfigState(),
+            customer: getCustomerState(),
+            cart: getCartState(),
+            paymentMethods: getPaymentMethodsState(),
+        });
+
+        jest.spyOn(store, 'dispatch').mockReturnValue(Promise.resolve(store.getState()));
+
+        paymentMethodMock = getMasterpass();
+        stripePaymentMethodMock = getStripe();
+
+        jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(paymentMethodMock);
+
+        const checkoutValidator = new CheckoutValidator(new CheckoutRequestSender(createRequestSender()));
+        orderActionCreator = new OrderActionCreator(orderRequestSender, checkoutValidator);
+        const paymentRequestSender = new PaymentRequestSender(createPaymentClient());
+        paymentActionCreator = new PaymentActionCreator(paymentRequestSender, orderActionCreator);
+        paymentMethodActionCreator = new PaymentMethodActionCreator(new PaymentMethodRequestSender(requestSender));
+
+        scriptLoader = new MasterpassScriptLoader(createScriptLoader());
+        masterpassScript = getMasterpassScriptMock();
+        jest.spyOn(scriptLoader, 'load').mockReturnValue(Promise.resolve(masterpassScript));
+        jest.spyOn(masterpassScript, 'checkout').mockReturnValue(true);
+
+        // Strategy
+        strategy = new MasterpassPaymentStrategy(
+            store,
+            orderActionCreator,
+            paymentActionCreator,
+            paymentMethodActionCreator,
+            scriptLoader
+        );
+    });
+
+    describe('#initialize()', () => {
+        it('throws an exception if payment method cannot be found', () => {
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(undefined);
+            const error = 'Unable to proceed because payment method data is unavailable or not properly configured.';
+            expect(() => strategy.initialize(initOptions)).toThrowError(error);
+        });
+
+        describe('without payment data', () => {
+            beforeEach(() => {
+                paymentMethodMock.initializationData = {
+                    allowedCardTypes: ['visa', 'amex', 'mastercard'],
+                    checkoutId: 'checkout-id',
+                };
+            });
+
+            it('throws an exception if checkout data is missing', () => {
+                jest.spyOn(store.getState().checkout, 'getCheckout').mockReturnValue(null);
+                const error = 'Unable to proceed because checkout data is unavailable.';
+                expect(() => strategy.initialize(initOptions)).toThrow(error);
```

Comment:
Instead of expecting the Error's Message, you can expect the error type:
IE:
``` typescript
expect(() => strategy.initialize(initOptions)).toThrow(InvalidArgumentError);
```

Reply:
you can also use a 'try catch' block to test the error and expect the error type instead of a string
```typescript
            it('throws an exception if checkout data is missing', async () => {
                jest.spyOn(store.getState().checkout, 'getCheckout').mockReturnValue(null);
                try {
                    await strategy.initialize(initOptions);
                } catch (e) {
                    expect(e).toBeInstanceOf(MissingDataError);
                }
            });
```